src/python/petab_sciml/problem_utils/neural_ode.py

```diff
@@ -0,0 +1,178 @@
+from libsbml import parseL3Formula, SBMLDocument, UNIT_KIND_SECOND, writeSBML
+import pandas as pd
+from typing import Iterable
+from ruamel.yaml import YAML
+
+def generate_neural_ode_problem(
```

Comment:
Is the target audience for this function, people who have never used PEtab before? If so then looks OK overall. If not, then I would prefer a different design

Reply:
Ah, this is a duplicate comment of what we discussed (i.e., the basic function will just accept a pytorch model + list of species IDs).

For experienced PEtab users, this might be cumbersome -- they might prefer the modular design we discussed.

Reply:
Python does support overloads
https://typing.python.org/en/latest/spec/overload.html

But the simpler approach is probably to just use `isinstance`.